infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -704,6 +720,10 @@ public void shutdown() {
         if(clusterManagerAdapter != null) {
             clusterManagerAdapter.shutdown();
         }
+
+        if (lock != null) {
```

Comment:
Will it cleaner to call lock.shutdown in the server shutdown call?

Reply:
That would mean exposing it. And the lock is really created by the Discovery Service, I think keeping it to the DiscoveryService scope is fine.